### DISCUSSION TITLE
Can´t configure Magnetometer to QMC5883L nor HC5983

### DISCUSSION THREAD

**Discussion by zark0-UwU** [iNavFlight/inav#7603] [Q&A]
Hi i know this issue is closed, bu nor version 3.0.1 nor 3.0.2 seem so allow me to select QMC5883L nor HMC5983 (Radiolink says they are identical) for magnetometer, and it does not work in any of the ones it allows me to select.
I tried via console, to see if its just a problem of the inav-configurator, but no, it also doesn't allow me to change it to neither of the two compasses.

---
A bit of background: My magnetometer is the one that comes with the  board of the gps [Radiolink TS100](https://www.radiolink.com/se100), and its hooked up to my MATEK H743-wing

_Originally posted by @zark0-UwU in https://github.com/iNavFlight/inav/issues/2801#issuecomment-942726605_

**Comment by JulioCesarMatias** [comment]
are your connections correct? You must use the SDA and SCL of I2C_1 of your FC. Please review your connections and tell me where they are

https://www.radiolink.com/se100

**Comment by zark0-UwU** [comment]
Im currently using the following wiring:
| gps/compass module | --> | Matek H743 |  
|-----------------------------|------|------------------|
|SDA(yellow)                | --> | DA1              |  
|SCL(green)                 | --> | CL1              | 


And the resources, as the come in the last FW for this board:
```
# resource
IO:
----------------------
A00: MOTOR3 OUT
A01: MOTOR4 OUT
A02: MOTOR5 OUT
A03: MOTOR6 OUT
A04: FREE 
A05: SPI1 SCK
A06: SPI1 MISO
A07: FREE 
A08: FREE 
A09: SERIAL1 UART TX
A10: SERIAL1 UART RX
A11: USB IN
A12: USB OUT
A13: FREE 
A14: FREE 
A15: BEEPER OUT
B00: MOTOR1 OUT
B01: MOTOR2 OUT
B02: MPU EXTI
B03: FREE 
B04: FREE 
B05: FREE 
B06: I2C1 SCL
B07: I2C1 SDA
B08: FREE 
B09: FREE 
B10: I2C2 SCL
B11: I2C2 SDA
B12: OSD CS
B13: SPI2 SCK
B14: SPI2 MISO
B15: SPI2 MOSI
C00: ADC CH1
C01: ADC CH2
C02: FREE 
C03: FREE 
C04: FREE 
C05: FREE 
C06: FREE 
C07: SERIAL6 UART RX
C08: FREE 
C09: FREE 
C10: FREE 
C11: FREE 
C12: FREE 
C13: FREE 
C14: FREE 
C15: MPU CS
D00: FREE 
D01: FREE 
D02: FREE 
D03: FREE 
D04: FREE 
D05: FREE 
D06: SERIAL2 UART RX
D07: SPI1 MOSI
D08: FREE 
D09: FREE 
D10: PINIO1 OUT
D11: PINIO2 OUT
D12: MOTOR7 OUT
D13: FREE 
D14: FREE 
D15: FREE 
E00: FREE 
E01: FREE 
E02: SPI_PREINIT CS
E03: LED1 OUT
E04: LED2 OUT
E05: FREE 
E06: FREE 
E07: FREE 
E08: FREE 
E09: FREE 
E10: FREE 
E11: SPI_PREINIT CS
E12: FREE 
E13: FREE 
E14: FREE 
E15: FREE 
```

**Comment by JohnWonFly** [comment]
hey friend,
if you bought ts100 from china, you can try to change rx tx by places.
sometimes they made wrong marks on board. 
just i forgot how we fixed this problem,
maybe bought new radiolink by HMC chip.

**Comment by JohnWonFly** [comment]
just i made drone by omnibus flight controller and mateksys pdb. 
it needs to power led light, motor regulators, beacon because we didnt want to load flight controller

**Comment by zark0-UwU** [comment]
Hi @JohnWonFly the ts100 is original, as far as i know, although i bought it on Bangood.
As far as i know the rx and tx cables are only used for GPS, wich is working great, its the magnetometer that is giving me trouble, wich is connected to the fc through SDA and SCL cables.
Please Let me kow if im wrong about anything i said.
And just so you know i have ny rx tx wires connected like the folowing:
| gps/compass module | --> | Matek H743 |  
|-----------------------------|------|------------------|
|RX(orange)                | --> | Rx2              |  
|TX(white)                 | --> | Tx2              |

**Comment by rwindsorjr** [comment]
The RX is to go to the TX and TX to RX or usually wont work. I bet the wire order is off on both the SDA SCL and RX TX
